sympy/external/gmpy.py

```diff
@@ -143,3 +151,41 @@ def invert(x, m):
             return pow(x, -1, m)
         except ValueError:
             raise ZeroDivisionError("invert() no inverse exists")
+
+    def legendre(x, y):
```

Comment:
I think we should make a new `sympy.external.ntheory` module and move all of the functions defined here inside there. Ideally this module should not have any nontrivial code and should just be used for selecting whether to use the gmpy2 versions of functions or otherwise local or mpmath versions.

Reply:
Is this what you are saying?

`sympy.external.gmpy`
```python
if gmpy is not None:
    ...
    gcd = gmpy.gcd
    lcm = gmpy.lcm
    invert = gmpy.invert
    legendre = gmpy.legendre
    jacobi = gmpy.jacobi
else:
    ...
    gcd = math.gcd
    lcm = math.lcm
    invert = sympy.external.ntheory.invert
    legendre = sympy.external.ntheory.legendre
    jacobi = sympy.external.ntheory.jacobi
```

`sympy.external.ntheory`
```python
def invert(x, m):
    return pow(x, -1, m)

def legendre(x, y):
    ...

def jacobi(x, y):
    ...
```

Reply:
I understand. Should PR be merged as it is and a new PR be created?